sns/cm/repair/ut/xform.c

```diff
@@ -438,6 +438,87 @@ static void cp_multi_failures_post(char data, int cnt, int index)
 	m0_semaphore_down(&sem);
 }
 
+static void buf_initialize(struct m0_buf *buf, uint32_t size, uint32_t len)
+{
+	uint32_t j;
+	uint8_t	*arr;
+
+	for (j = 0; j < size; ++j) {
+		M0_ALLOC_ARR(arr, len);
+		M0_UT_ASSERT(arr != NULL);
+		m0_buf_init(&buf[j], arr, len);
+	}
+}
+
+static void buf_free(struct m0_buf *buf, uint32_t count)
+{
+	uint32_t i;
+
+	for (i = 0; i < count; ++i)
+		m0_buf_free(&buf[i]);
+}
+
+static void bufvec2buf(struct m0_buf *buf, struct m0_bufvec *bufvec, uint32_t count)
+{
+	uint32_t  i;
+	uint32_t  j;
+	uint8_t  *buf_data;
+	uint8_t  *bufvec_data;
+
+	for (i = 0; i < count; i++) {
+		buf_data = (uint8_t *)buf[i].b_addr;
+		for (j = 0; j < SEG_NR; ++j) {
+			bufvec_data = (uint8_t *)bufvec[i].ov_buf[j];
+			memcpy(&buf_data[SEG_SIZE * j], bufvec_data, SEG_SIZE);
+		}
+	}
+}
+
+static void buf2bufvec(struct m0_bufvec *bufvec, struct m0_buf *buf, uint32_t count)
+{
+	uint32_t  i;
+	uint32_t  j;
+	uint8_t  *buf_data;
+	uint8_t  *bufvec_data;
+
+	for (i = 0; i < count; i++) {
+		buf_data = (uint8_t *)buf[i].b_addr;
+		for (j = 0; j < SEG_NR; ++j) {
+			bufvec_data = (uint8_t *)bufvec[i].ov_buf[j];
+			memcpy(bufvec_data, &buf_data[SEG_SIZE * j], SEG_SIZE);
+		}
+	}
+}
+
+static void ref_parity_calculate(struct m0_parity_math *math, struct m0_bufvec *bufvec)
+{
+	struct m0_buf *buf;
+	uint32_t       unit_count;
+
+	M0_UT_ASSERT(math != NULL);
+	M0_UT_ASSERT(bufvec != NULL);
+
+	unit_count = math->pmi_data_count + math->pmi_parity_count;
+
+	M0_ALLOC_ARR(buf, unit_count);
+	M0_UT_ASSERT(buf != NULL);
```

Comment:
if we don't care the stack size, can we:
```
	uint32_t       unit_count = math->pmi_data_count + math->pmi_parity_count;;
	struct m0_buf buf[unit_count];

```
I remember you used this "variable sized array" before.

Current implementation is fine. But this one is with less code.
I prefer to rename "buf" to "buf_array" or "buf_arr", or "bufs", to say it is not a single m0_buf, but an array.

Reply:
Done.